File: DataAccessLayerWriter/Field.cs
using System;

namespace DataAccessLayerWriter
{
    public interface IType
    {
        String Name { get; set; }
        DataType Type { get; set; }
    }

    public class Field : IType
    {
        public String Name { get; set; }
        public DataType Type { get; set; }
        public Boolean AllowsNull { get; set; }

        public int? Length { get; set; }

        public bool IsOutput { get; set; }
    }
}

File: DataAccessLayerWriterToFSharp/FSharp/Field.fs
namespace DataAccessLayerWriter

    open System;

    type IType = {
        Name:string;
        DataType:Type
    }

    type Field = {
         Name:string; 
         Type:DataType; 
         AllowsNull:bool;
         Length:int option;
         IsOutput:bool
    }
